### PR TITLE
CAP-29 Implemented user auth for frontend; profile route for backend

### PR DESCRIPTION
**Pull request by snaanortey**
Please review this PR which has the summary below:

### Backend
- Created `profile route API` to return user profile once log in is successful

### Frontend
- Created `SignUp` page for user registration
- Created `LogIn` page for user login using `email` and `password`
- Created `Profile` page for displaying user profile upon successful login

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "@types/react": "^18.0.0",
     "@types/react-dom": "^18.0.0",
     "axios": "^1.1.3",
+    "axios-hooks": "^4.0.0",
```
Comment (ehimah) [inline client/package.json:18]: clean up

File: client/src/services/backend.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+   * @param route a / prefixed route segment e.g. /profile
+   * @returns a full backend API url
+   */
+  private buildFullUrl = (route: string): string => {
+    return `${this.baseUrl}${route}`;
+  };
+
+  public login = async (
+    userCredentials: LoginCredentials
+  ): Promise<boolean> => {
```
Comment (ehimah) [inline client/src/services/backend.ts:36]: add ts-docs comments

File: client/src/services/backend.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      // Save token to bearerToken property
+      this.bearerToken = token;
+      return true;
+    } catch (err) {
+      console.log(err);
+      return false;
+    }
+  };
+
+  public getUserProfile = async (): Promise<UserProfile> => {
```
Comment (ehimah) [inline client/src/services/backend.ts:57]: add ts-docs comments

**Review by ehimah — APPROVED** [review]
LGTM

address the few comments and it's ✅

File: client/src/services/backend.ts (hunk 2 above)
Comment (snaanortey) [reply client/src/services/backend.ts:36]: Addressed this. thanks!

File: client/src/services/backend.ts (hunk 3 above)
Comment (snaanortey) [reply client/src/services/backend.ts:57]: added

**Pull request merged**